test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1367 @@
+		await paragraphBlock.click( {
+			position: { x: 20, y: height / 2 },
+			modifiers: [ 'Shift' ],
+		} );
+		await page.keyboard.type( 'hi' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/group',
+				innerBlocks: [
+					{
+						name: 'core/paragraph',
+						attributes: { content: 'hih text in group' },
+					},
+				],
+			},
+		] );
+	} );
+
+	test( 'should select by dragging', async ( {
+		page,
+		editor,
+		multiBlockSelectionUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '2' },
+		} );
+		// To hide the block tool bar.
+		await page.keyboard.press( 'ArrowDown' );
+
+		const [ paragraph1, paragraph2 ] = await editor.canvas
+			.getByRole( 'document', { name: 'Paragraph block' } )
+			.all();
+
+		await paragraph1.hover();
+		await page.mouse.down();
+		await paragraph2.hover();
+		await page.mouse.up();
+
+		await expect( multiBlockSelectionUtils.assertNativeSelection ).toPass();
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedFlatIndices )
+			.toEqual( [ 1, 2 ] );
+	} );
+
+	test( 'should select by dragging into nested block', async ( {
+		page,
+		editor,
+		multiBlockSelectionUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/group',
+			innerBlocks: [
+				{
+					name: 'core/paragraph',
+					attributes: { content: '2' },
+				},
+			],
+		} );
+		// To hide the block tool bar.
+		await page.keyboard.press( 'ArrowDown' );
+		await page.keyboard.press( 'ArrowDown' );
+
+		const [ paragraph1, paragraph2 ] = await editor.canvas
+			.getByRole( 'document', { name: 'Paragraph block' } )
+			.all();
+
+		await paragraph1.hover();
+		await page.mouse.down();
+		await paragraph2.hover();
+		await page.mouse.up();
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedFlatIndices )
+			.toEqual( [ 1, 2 ] );
+	} );
+
+	test( 'should cut and paste', async ( { editor, pageUtils } ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '2' },
+		} );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'x' );
+
+		await expect.poll( editor.getBlocks ).toEqual( [] );
+
+		await pageUtils.pressKeyWithModifier( 'primary', 'v' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: '1' } },
+			{ name: 'core/paragraph', attributes: { content: '2' } },
+		] );
+	} );
+
+	test( 'should copy and paste', async ( { page, editor, pageUtils } ) => {
```

Comment:
As per the [best practices](https://playwright.dev/docs/best-practices#write-fewer-tests-but-longer-tests) of Playwright, this is now a combination of multiple similar test cases:

- `should copy and paste`
- `should place the caret at the end of last pasted paragraph (paste to empty editor)`
- `should place the caret at the end of last pasted paragraph (paste mid-block)`
- `should place the caret at the end of last pasted paragraph (replace)`